### PR TITLE
Fix should help examples

### PR DESCRIPTION
**Pull request by asears**
Signed-off-by: asears <asears@users.noreply.github.com>

## 1. General summary of the pull request

Improves #1634

### CODE REVIEW

File: src/functions/assertions/Should.ps1 [hunk 1]
```diff
@@ -22,23 +22,76 @@ function New-ShouldErrorRecord ([string] $Message, [string] $File, [string] $Lin
...
-    need to be verified. Each Should keywords need to be located in a new line.
+    need to be verified. Each Should keyword needs to be on a separate line.
     Test will be passed only when all assertion will be met (logical conjuction).
 
     .LINK
-    https://github.com/pester/Pester/wiki/Should
+    https://pester.dev/docs/usage/assertions
 
     .LINK
     about_Should
```
Comment (asears) [inline src/functions/assertions/Should.ps1:40]: Are these links still valid, how can I test?

File: BUILD.md [hunk 2]
```diff
@@ -1,13 +1,48 @@
...
+
+Afterwards, each test can be run individually using Invoke-Pester.
+
+Test.ps1 and optionally -skipPTests to skip the .ts.ps1 files.
+
+## test.ps1
+
+test.ps1 can be run with the following parameters
+
+TODO document parameters, tests executed in pipelines.
```
Comment (asears) [inline BUILD.md:30]: Need some additional help for the parameters used in test.ps1

**Comment by bravo-kernel** [comment]
Very nice PR mr. @asears. I personally really appreciate these kind of changes for their educational value.

I've added one remark relating to the multi line code examples, they should be fenced or they will not render properly on the pester.dev website. https://docusaurus-powershell.netlify.app/docs/faq/multi-line-examples

File: BUILD.md [hunk 3]
```diff
@@ -1,13 +1,48 @@
-## Building Pester
+# Building Pester
+
+Pester is written in Powershell and C#.  The Microsoft .NET 4.5.2 Framework is required to build the Pester binaries.
```
Comment (nohwnd) [inline BUILD.md:3]:
```suggestion
Pester is written in Powershell and C#.  The Microsoft .NET Framework 4.5.2 is required to build the Pester binaries.
```

File: BUILD.md [hunk 4]
```diff
@@ -1,13 +1,48 @@
...
 
 Pester has a C# Solution which requires .Net Framework SDKs and Developer Packs in order to compile. The targetted frameworks can be found in `src\csharp\Pester\Pester.csproj`.
```
Comment (nohwnd) [inline BUILD.md:5]:
```suggestion
Pester has a C# Solution which requires .NET Framework SDKs and Developer Packs in order to compile. The targeted frameworks can be found in `src\csharp\Pester\Pester.csproj`.
```

File: BUILD.md [hunk 5]
```diff
@@ -1,13 +1,48 @@
...
+
+TODO document parameters, tests executed in pipelines.
+
+```powershell
+.\test.ps1 -CI -SkipPTests -NoBuild -File ${filename}
+```
+
+## Continuous Integration
+
+The Azure Devops Pipeline azure-pipelines.yml file contains the code used for builds, unit and integration tests.
```
Comment (nohwnd) [inline BUILD.md:38]:
```suggestion
The Azure Devops Pipeline azure-pipelines.yml file contains the code used for builds, unit and integration tests in the CI pipeline.
```

File: BUILD.md [hunk 6]
```diff
@@ -1,13 +1,48 @@
...
+
+## Documentation
+
+Documentation is available in the repo, the wiki, and at <https://pester.dev>
+
+Documentation is written in Markdown. Comment-based Documentation and parts of the documentation web site are generated using Docusaurus Powershell.
```
Comment (nohwnd) [inline BUILD.md:44]:
```suggestion
Documentation is written in Markdown. Comment-based Documentation and parts of the documentation web site are generated using Docusaurus PowerShell.
```

File: BUILD.md [hunk 7]
```diff
@@ -1,13 +1,48 @@
...
+
+Multi-line Examples added to comments should use fenced code.
```
Comment (nohwnd) [inline BUILD.md:46]:
```suggestion
Multi-line examples added to comments should use fenced code.
```

**Review by nohwnd — COMMENTED** [review]
Few suggestions.

File: src/functions/assertions/Should.ps1 [hunk 8]
```diff
@@ -22,23 +22,76 @@ function New-ShouldErrorRecord ([string] $Message, [string] $File, [string] $Lin
...
     about_Pester
+
+    .EXAMPLE
+    Describe "d1" {
```
[the remark this answers is not in the document]
Comment (asears) [reply src/functions/assertions/Should.ps1:47]: Can you confirm these are fixed or identify way to test?

File: BUILD.md (hunk 2 above)
Comment (nohwnd) [reply BUILD.md]: param (
    # forces P Tests  to fail when I leave `dt` in the tests. dt only runs the specified test, so leaving it in code would run only one test from the file on the server. Also enable exit with 1 if tests don't pass
    [switch]$CI,
    # skip the tests written using the P module, those are unit tests for the runtime, and acceptance tests for whole Pester
    [switch]$SkipPTests,
    # do not build the underlying csharp part, used in CI where we already built it from clean
    [switch]$NoBuild,
    # pass the file to run Pester (not P) tests from
    [string[]] $File
)

Comment (nohwnd) [reply BUILD.md]: Ah you already described it.

File: src/functions/assertions/Should.ps1 (hunk 8 above)
[the remark this answers is not in the document]
Comment (bravo-kernel) [reply src/functions/assertions/Should.ps1:47]: @asears I saw that you added the code fences and that should render perfectly. Thanks a lot, much appreciated!

File: BUILD.md [hunk 9]
```diff
@@ -1,13 +1,64 @@
...
+Afterwards, each test can be run individually using Invoke-Pester.
+
+Test.ps1 and optionally -skipPTests to skip the .ts.ps1 files.
+
+## test.ps1
+
+test.ps1 can be run with the following parameters:
+
+```powershell
+.\test.ps1 -CI -SkipPTests -NoBuild -File ${filename}
```
[the remark this answers is not in the document]
Comment (asears) [reply BUILD.md]: Fixed

**Comment by nohwnd** [comment]
LGTM, @asears  are you done with all your changes?

**Comment by asears** [comment]
Yes will open another PR if I have further improvements.

**Pull request merged**